Runtime/Native/Windows/NativeClient.cs

```diff
@@ -0,0 +1,438 @@
+                        // we won't false positive ANR report
+                        lastUpdatedCache = 0;
+                    }
+                    Thread.Sleep(5000);
+                }
+            });
+            _anrThread.IsBackground = true;
+            _anrThread.Start();
+            return;
+        }
+
+        public bool OnOOM()
+        {
+            return false;
+        }
+
+        public void PauseAnrThread(bool state)
+        {
+            _preventAnr = state;
+        }
+
+        public void SetAttribute(string key, string value)
+        {
+            if (string.IsNullOrEmpty(key))
+            {
+                return;
+            }
+            // avoid null reference in crashpad source code
+            if (value == null)
+            {
+                value = string.Empty;
+            }
+            AddAttributes(key, value);
+        }
+
+        public void UpdateClientTime(float time)
+        {
+            _lastUpdateTime = time;
+        }
+
+        /// <summary>
+        /// Read directory structure in the native crash directory and send new crashes to Backtrace
+        /// </summary>
+        public static IEnumerator SendUnhandledGameCrashesOnGameStartup(IEnumerable<string> attachments, IBacktraceApi backtraceApi)
+        {
+            // Path to the native crash directory
+            string nativeCrashesDir = Path.Combine(
+                    Environment.GetFolderPath(Environment.SpecialFolder.LocalApplicationData),
+                    string.Format("Temp/{0}/{1}/crashes", Application.companyName, Application.productName));
+
+            if (string.IsNullOrEmpty(nativeCrashesDir) || !Directory.Exists(nativeCrashesDir))
+            {
+                yield break;
+            }
+            var crashDirs = Directory.GetDirectories(nativeCrashesDir);
+            foreach (var crashDir in crashDirs)
+            {
+
+                var crashDirFullPath = Path.Combine(nativeCrashesDir, crashDir);
+                var crashFiles = Directory.GetFiles(crashDirFullPath);
+
+                var alreadyUploaded = crashFiles.Any(n => n.EndsWith("backtrace.json"));
+                if (alreadyUploaded)
+                {
+                    continue;
+                }
+                var minidumpPath = crashFiles.FirstOrDefault(n => n.EndsWith("crash.dmp"));
+                if (string.IsNullOrEmpty(minidumpPath))
+                {
+                    continue;
+                }
+                if (attachments == null)
+                {
+                    attachments = new List<string>();
+                }
+                attachments.Concat(crashFiles.Where(n => n != minidumpPath));
+                IDictionary<string, string> attributes = GetScopedAttributes();
+                // be sure that error.type attribute provided by default by our library
+                // is always present in native attributes.
+                attributes["error.type"] = "Crash";
```

Comment:
Should we check previous value of `attributes["error.type"]` so we don't overwrite something in case we have a Hang or some other kind of `error.type` here? Or are these only dumps generated by Unity in which case it's safe?

Reply:
no - every minidump file in this directory is a crash.